src/components/AttachmentModal.tsx

```diff
@@ -1,189 +1,211 @@
-    source: PropTypes.oneOfType([PropTypes.string, sourcePropTypes]),
+type AttachmentModalOnyxProps = {
+    /** The transaction associated with the receipt attachment, if any */
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
 
-    /** Optional callback to fire when we want to preview an image and approve it for use. */
-    onConfirm: PropTypes.func,
+    /** The report associated with the receipt attachment, if any */
+    parentReport: OnyxEntry<OnyxTypes.Report>;
 
-    /** Whether the modal should be open by default */
-    defaultOpen: PropTypes.bool,
+    /** The policy associated with the receipt attachment, if any */
+    policy: OnyxEntry<OnyxTypes.Policy>;
 
-    /** Optional callback to fire when we want to do something after modal show. */
-    onModalShow: PropTypes.func,
+    /** The list of report actions associated with the receipt attachment, if any */
+    parentReportActions: OnyxEntry<OnyxTypes.ReportActions>;
+};
 
-    /** Optional callback to fire when we want to do something after modal hide. */
-    onModalHide: PropTypes.func,
+type Attachment = {
+    source: string;
+    isAuthTokenRequired: boolean;
+    file: File;
+    isReceipt: boolean;
+};
 
-    /** Optional callback to fire when we want to do something after attachment carousel changes. */
-    onCarouselAttachmentChange: PropTypes.func,
+type ImagePickerResponse = {
+    height: number;
+    name: string;
+    size: number;
+    type: string;
+    uri: string;
+    width: number;
+};
 
-    /** Optional original filename when uploading */
-    originalFileName: PropTypes.string,
+type FileObject = File | ImagePickerResponse;
 
-    /** A function as a child to pass modal launching methods to */
-    children: PropTypes.func,
+type ChildrenProps = {
+    displayFileInModal: (data: FileObject) => void;
+    show: () => void;
+};
+
+type AttachmentModalProps = AttachmentModalOnyxProps & {
+    /** Optional source (URL, SVG function) for the image shown. If not passed in via props must be specified when modal is opened. */
+    source?: AvatarSource;
+
+    /** Optional callback to fire when we want to preview an image and approve it for use. */
+    onConfirm?: ((file: Partial<File>) => void) | null;
+
+    /** Whether the modal should be open by default */
+    defaultOpen?: boolean;
+
+    /** Optional original filename when uploading */
+    originalFileName?: string;
 
     /** Whether source url requires authentication */
-    isAuthTokenRequired: PropTypes.bool,
+    isAuthTokenRequired?: boolean;
 
     /** Determines if download Button should be shown or not */
-    allowDownload: PropTypes.bool,
+    allowDownload?: boolean;
 
     /** Title shown in the header of the modal */
-    headerTitle: PropTypes.string,
+    headerTitle?: string;
 
     /** The report that has this attachment */
-    report: reportPropTypes,
+    report?: OnyxTypes.Report;
```

Comment:
```suggestion
    report?: OnyxEntry<OnyxTypes.Report>;
```

I believe we are passing a onyx value to it